models/dojo.py

```diff
@@ -0,0 +1,277 @@
+import random
+
+from models.room.office_space import Office
+from models.room.living_space import LivingSpace
+from models.person.fellow import Fellow
+from models.person.staff import Staff
+
+
+class Dojo(object):
+    """The Dojo class."""
+
+    office_rooms = {}
+    living_rooms = {}
+    persons = {}
+    divider = ("\n{}\n".format("-" * 30))
+
+    def __init__(self):
+        pass
+
+    def create_room(self, arg):
+        """The function creates a new room."""
+        room_type = arg["<room_type>"]
+        room_names = arg["<room_name>"]
+        for room_name in room_names:
+            if room_name == ('office' or 'livingspace'):
+                raise Exception("This is not a valid room name")
+            else:
+                if room_type.upper() == "OFFICE":
+                    self.office_rooms[room_name] = Office(room_name)
+
+                elif room_type.upper() == "LIVINGSPACE":
+                    self.living_rooms[room_name] = LivingSpace(room_name)
+
+                else:
+                    raise Exception("Invalid Room Type.Must be office or living")
+
+                Prefix = ("A" if room_type.upper() == "LIVINGSPACE" else "An")
+                print("{} {} called {} has been successfully created".format(
+                    Prefix, room_type, room_name))
+
+    def add_person(self, arg):
+        """The function adds a person to a room randomly."""
+        first_name = arg["<first_name>"]
+        last_name = arg["<last_name>"]
+        rank = (arg["<FELLOW/STAFF>"])
+        wants_accomodation = (arg["<wants_accomodation>"] or "N")
+
+        if rank.upper() == "STAFF":
+            new_user = Staff(first_name, last_name)
+            staff_id = "S" + str(new_user.id)
+            self.persons[staff_id] = new_user
+
+            print("{0} {1} has been successfully added".format(self.persons[
+                staff_id].rank, self.persons[staff_id]))
+
+        elif rank.upper() == "FELLOW":
+            new_user = Fellow(first_name, last_name,
+                              wants_accomodation.upper())
+            fellow_id = "F" + str(new_user.id)
+            self.persons[fellow_id] = new_user
+
+            print("{0} {1} has been successfully added".format(self.persons[
+                fellow_id].rank, self.persons[fellow_id]))
+
+        else:
+            raise Exception('Person can only be a fellow or staff')
+
+        self.assign_person(new_user)
+
+    @staticmethod
+    def get_available_room_spaces(room_spaces):
+        available_room_spaces = []
+        for room, room_info in room_spaces.items():
```

Comment:
this looks like a good candidate to use python's list comprehensions
see [here](http://stackoverflow.com/a/15474969/6139947)